app/src/main/java/com/p2p/presentation/tuttifrutti/review/TuttiFruttiReviewRoundAdapter.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.viewbinding.ViewBinding
+import com.p2p.R
+import com.p2p.databinding.ViewReviewCategoryItemBinding
+import com.p2p.databinding.ViewReviewWordItemBinding
+import com.p2p.model.tuttifrutti.FinishedRoundInfo
+import com.p2p.model.tuttifrutti.FinishedRoundPointsInfo
+import com.p2p.presentation.tuttifrutti.create.categories.Category
+import com.p2p.utils.isEven
+
+/** The adapter used to show the list of round reviews. */
+class TuttiFruttiReviewRoundAdapter(private val onChangeRoundPoints: (String, String, Int) -> Unit) :
+    RecyclerView.Adapter<TuttiFruttiReviewRoundAdapter.RecyclerViewHolder>() {
+
+    var finishedRoundInfo = listOf<FinishedRoundInfo>()
+    var finishedRoundPointsInfo = listOf<FinishedRoundPointsInfo>()
+        set(value) {
+            field = value
+            notifyDataSetChanged()
+        }
+
+    override fun getItemViewType(position: Int): Int {
+        // If the the modulus number of players plus one is zero, then we get the category title view
+        return position % (finishedRoundInfo.count() + 1)
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
+        if (viewType == 0) {
+            return ReviewCategoryViewHolder(
+                ViewReviewCategoryItemBinding.inflate(
+                    LayoutInflater.from(parent.context),
+                    parent,
+                    false
+                )
+            )
+        }
+        return ReviewWordViewHolder(
+            ViewReviewWordItemBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
+        val viewIndex = getItemViewType(position)
+        val viewHolderParams = RecyclerViewHolderParameters()
+        val categoryIndex = getCategoryIndex(position)
+        val viewCategory = getCategory(categoryIndex)
+
+        if (viewIndex == 0) {
+            viewHolderParams.category = viewCategory
+        } else {
+            viewHolderParams.player = finishedRoundInfo[viewIndex - 1].player
+            viewHolderParams.word = finishedRoundInfo[viewIndex - 1].categoriesWords[viewCategory]!!
+            viewHolderParams.points = finishedRoundPointsInfo.find {
+                it.player == viewHolderParams.player
+            }!!.wordsPoints[categoryIndex]
+        }
+
+        return holder.bind(viewHolderParams, position)
+    }
+
+    // Return the number of players multiplied by the number of categories plus the number of categories (for the titles)
+    override fun getItemCount() = finishedRoundInfo.count() * numberOfCategories() + numberOfCategories()
+
+    private fun numberOfCategories() = finishedRoundInfo.first().categoriesWords.count()
+
+    private fun getCategory(categoryIndex: Int): Category {
+        return finishedRoundInfo.first().categoriesWords.keys.toTypedArray()[categoryIndex]
```

Comment:
keys es un set, me da miedo que le cambie el orden a las categories, sería mejor capaz:
```suggestion
        return finishedRoundInfo.first().categoriesWords.toList()[categoryIndex].first
```

Reply:
De uni 🤗